src/typedefs.jl

```diff
@@ -73,7 +75,7 @@ end
 
 ## Nullable Arrays
 
-@compat abstract type AbstractNullableCategoricalArray{T, N, R} <: AbstractArray{Nullable{CategoricalValue{T, R}}, N} end
+@compat abstract type AbstractNullableCategoricalArray{T, N, R} <: AbstractArray{Union{CategoricalValue{T, R}, Null}, N} end
```

Comment:
I've just pushed a new commit which passes most tests and should be OK from an API point of view, but needs more work to clean many things.